cmd/plume/prune.go

```diff
@@ -0,0 +1,194 @@
+		}
+
+		for _, key := range *storageKey.Keys {
+			blobs, err := api.ListBlobs(spec.Azure.StorageAccount, *key.Value, container, storage.ListBlobsParameters{})
+			if err != nil {
+				plog.Warningf("Error listing blobs: %v", err)
+			}
+			plog.Infof("Got %d blobs for container %q (key %v)\n", len(blobs), container, key)
+
+			now := time.Now()
+			for _, blob := range blobs {
+				lastModifiedDate, err := time.Parse(time.RFC1123, blob.Properties.LastModified)
+				if err != nil {
+					plog.Warningf("Error converting last modified date (%v): %v", blob.Properties.LastModified, err)
+				}
+				duration := now.Sub(lastModifiedDate)
+				daysOld := int(duration.Hours() / 24)
+				if daysOld >= days {
+					plog.Infof("Old blob %q: %d days old\n", blob.Name, daysOld)
+					if !pruneDryRun {
+						plog.Infof("Deleting Blob %q in container %q", blob.Name, container)
+						err = api.DeleteBlob(spec.Azure.StorageAccount, *key.Value, container, blob.Name)
+						if err != nil {
+							plog.Warningf("Error deleting blob (%v): %v", blob.Name, err)
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+func pruneAWS(ctx context.Context, spec *channelSpec) {
+	if spec.AWS.Image == "" || awsCredentialsFile == "" {
+		plog.Notice("AWS image pruning disabled.")
+		return
+	}
+
+	// Iterate over all partitions and regions in the given channel and prune
+	// images in each of them.
+	for _, part := range spec.AWS.Partitions {
+		for _, region := range part.Regions {
+			if pruneDryRun {
+				plog.Printf("Checking for images in %v...", part.Name)
+			} else {
+				plog.Printf("Pruning images in %v...", part.Name)
+			}
+
+			api, err := aws.New(&aws.Options{
+				CredentialsFile: awsCredentialsFile,
+				Profile:         part.Profile,
+				Region:          region,
+			})
+			if err != nil {
+				plog.Fatalf("Creating client for %v %v: %v", part.Name, region, err)
+			}
+
+			images, err := api.GetImagesByTag("Channel", specChannel)
+			if err != nil {
+				plog.Fatalf("Couldn't list images in channel %q: %v", specChannel, err)
+			}
+
+			plog.Infof("Got %d images with channel %q\n", len(images), specChannel)
+
+			now := time.Now()
+			for _, image := range images {
+				creationDate, err := time.Parse(time.RFC3339Nano, *image.CreationDate)
+				if err != nil {
+					plog.Warningf("Error converting creation date (%v): %v", *image.CreationDate, err)
+				}
+				duration := now.Sub(creationDate)
+				daysOld := int(duration.Hours() / 24)
+				if daysOld >= days {
+					plog.Infof("Old image %q: %d days old\n", *image.Name, daysOld)
+					if !pruneDryRun {
+						// Construct the s3ObjectPath in the same manner it's constructed for upload
+						var version string
+						for _, t := range image.Tags {
+							if *t.Key == "Version" {
+								version = *t.Value
+							}
+						}
+						imageFileName := spec.AWS.Image
+						s3ObjectPath := fmt.Sprintf("%s-usr/%s/%s", *image.Architecture, version, strings.TrimSuffix(imageFileName, filepath.Ext(imageFileName)))
```

Comment:
`*image.Architecture` returns `x86_64` instead of `amd64` and needs some postprocessing.

Reply:
Ok, I think I've fixed this although I don't think it matters because the S3 bucket is empty (I think whatever gets uploaded there gets deleted once it turns into a snapshot).